### PR TITLE
Feature/sc 105371 kubernetes 1 26

### PR DESCRIPTION
**Pull request by amandineslx**
Support the new authentication mode for `kubectl` commands in Gcloud.

This authentication plugin is already available but will be the only allowed authentication method starting with Kubernetes 1.26.

See [Kubernetes documentation](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke).

For this plugin version to work, the Google authentication plugin needs to be installed in the machine:

```bash
sudo su
sudo yum install google-cloud-sdk-gke-gcloud-auth-plugin
```

A PR will be opened in dip repository to have this module installed at machine startup.

To be merged after [PR removing the ABAC legacy auth](https://github.com/dataiku/dss-plugin-gke-clusters/pull/21) as this auth mode will be broken by the changes.

### CODE REVIEW

File: python-lib/dku_google/gcloud.py [hunk 1]
```diff
@@ -117,3 +118,44 @@ def get_instance_service_account():
...
+
+def create_kube_config_file(cluster_id, is_cluster_regional, kube_config_path):
+    """
+    Delegate the creation of the kube config file to gke-gcloud-auth-plugin
+    Starting with Kubernetes 1.26, the authentication to execute kubectl commands on Google clusters
+        won't be available in kubectl anymore and the client go auth plugin needs to be used
+    The gke-gcloud-auth-plugin is installed on the machine directly from the image
+    It will keep an authentication token linked to the unix user on the machine for gcloud and kubectl calls
+    Command `gcloud container clusters get-credentials CLUSTER_NAME` configures the authentication
+        for the specified cluster
```
Comment (jules-casoli) [inline python-lib/dku_google/gcloud.py:131]:
```suggestion
        for the specified cluster, and creates an adequate kubeconfig file.
```

**Review by jules-casoli — CHANGES_REQUESTED** [review]
Lacks the CHANGELOG.
Also, I didn't realise that legacy-auth was a `client-facing` option. Meaning it's removal is a bit more impacting than I though. Should we put it in a separate PR then ?

**Review by marchelleboid — CHANGES_REQUESTED** [review]
Tested with clusters on version 1.21 and 1.24 (oldest and newest available on GKE), and it works.

I'm wondering about the new step to install the "auth plugin". For DSS provisioned by FM, that's easy enough for add so it's transparent to the users, but what about users who aren't using FM? Can we make it clear enough for them that there's an additional manual step of installing the auth plugin, or else creating new clusters won't work with this version of the cluster plugin? I see that the auth plugin can be installed directly by a `gcloud` command, even if it's not the recommended way for Linux, could the cluster plugin itself try to do this? (for example in the EKS plugin, it installs `eksctl` if it can't find it)

**Comment by amandineslx** [comment]
> I'm wondering about the new step to install the "auth plugin". For DSS provisioned by FM, that's easy enough for add so it's transparent to the users, but what about users who aren't using FM? Can we make it clear enough for them that there's an additional manual step of installing the auth plugin, or else creating new clusters won't work with this version of the cluster plugin? I see that the auth plugin can be installed directly by a `gcloud` command, even if it's not the recommended way for Linux, could the cluster plugin itself try to do this? (for example in the EKS plugin, it installs `eksctl` if it can't find it)

The error message is quite explicit about the missing executable. However, we can add to the documentation about elastic setups that the user needs to make sure this plugin is installed on the DSS machine before trying to spin up the cluster.

About the ad hoc installation, `dataiku` user is not a sudoer on the machine and cannot run the `yum install` command documented in the installation guide. Also, `gcloud` installation command does not work as the component manager seems to be disabled on the machine.

**Comment by marchelleboid** [comment]
> The error message is quite explicit about the missing executable. However, we can add to the documentation about elastic setups that the user needs to make sure this plugin is installed on the DSS machine before trying to spin up the cluster.
> 
> About the ad hoc installation, `dataiku` user is not a sudoer on the machine and cannot run the `yum install` command documented in the installation guide. Also, `gcloud` installation command does not work as the component manager seems to be disabled on the machine.

That makes sense to me (even trying a yum install would be complicated because you would need to figure out if yum is even the correct package manager to use). It's true that in our documentation, we already tell the user that they must have `gcloud`, `docker`, and `kubectl` installed, so this auth plugin can also be added there

**Review by marchelleboid — APPROVED** [review]

**Comment by amandineslx** [comment]
@marchelleboid if you want to have a look at the changelog, it has been updated now :)

**Review by jules-casoli — APPROVED** [review]

**Review by marchelleboid — APPROVED** [review]

**Review by alex-dataiku — APPROVED** [review]
LGTM:
- successfully attached cluster in version `v1.23.12-gke.100`
- successfully created cluster in version `v1.25.4-gke.2100`

**Pull request merged**